csrc/prepare_inputs/advance_step.cu

```diff
@@ -0,0 +1,126 @@
+/*
+ * TODO: Add doc
+ */
+
+#include "advance_step.cuh"
+
+namespace prepare_inputs {
+
+template <int const num_threads>
+__global__ void advance_step_kernel(int num_seqs, int num_queries,
```

Comment:
add comment which attention backends this supports? I guess we should name it correspondingly as well ?

Reply:
I added docs in the file's header to give a bit more info. In general, supporting other attn backends will not require implementing other GPU kernels, since the logic is mostly shared. In the worst case, this GPU kernel should simply update more tensors or take into account of tensor reshapes.